src/WebJobs.Extensions.Http/Bindings/ClaimsIdentityBindingProvider.cs

```diff
@@ -0,0 +1,167 @@
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Http
+{
+    /// <summary>
+    /// This provider provides a binding to Type <see cref="ClaimsIdentity"/>.
+    /// </summary>
+    /// <remarks>
+    /// Attempts to bind to an identity in precedence order based on <see cref="AuthorizationLevel"/> levels.
+    /// I.e. if the request is key authenticated (Function/System/Admin) that identity is returned, if the
+    /// request is user authenticated (User) that identity is returned.
+    /// </remarks>
+    internal class ClaimsIdentityBindingProvider : IBindingProvider
+    {
+        public Task<IBinding> TryCreateAsync(BindingProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            if (context.Parameter.ParameterType != typeof(ClaimsIdentity))
+            {
+                return Task.FromResult<IBinding>(null);
+            }
+
+            return Task.FromResult<IBinding>(new ClaimsIdentityBinding(context.Parameter));
+        }
+
+        private class ClaimsIdentityBinding : IBinding
+        {
+            private readonly ParameterInfo _parameter;
+
+            public ClaimsIdentityBinding(ParameterInfo parameter)
+            {
+                _parameter = parameter;
+            }
+
+            public bool FromAttribute
+            {
+                get { return false; }
+            }
+
+            public async Task<IValueProvider> BindAsync(BindingContext context)
+            {
+                if (context == null)
+                {
+                    throw new ArgumentNullException("context");
+                }
+
+                // if the request is EasyAuth authenticated, get the identity provider type
+                object value = null;
+                string identityProvider = null;
+                if (context.BindingData.TryGetValue(HttpTriggerAttributeBindingProvider.HttpHeadersKey, out value))
+                {
+                    IDictionary<string, string> headers = (IDictionary<string, string>)value;
+                    if (headers != null)
+                    {
+                        headers.TryGetValue(HttpConstants.AntaresEasyAuthProviderHeaderName, out identityProvider);
+                    }  
+                }
+
+                var identity = GetPrimaryIdentity(ClaimsPrincipal.Current, identityProvider);
+                return await BindInternalAsync(identity);
+            }
+
+            public Task<IValueProvider> BindAsync(object value, ValueBindingContext context)
+            {
+                if (context == null)
+                {
+                    throw new ArgumentNullException("context");
+                }
+
+                var identity = GetPrimaryIdentity(ClaimsPrincipal.Current);
```

Comment:
Yes, this won't work in .NET Core because .NET Core doesn't play nice with IIS modules.  You would need to write some logic which manually creates the identity from the HTTP request headers.